src/ngraph/op/embedding_lookup.cpp

```diff
@@ -0,0 +1,58 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/embedding_lookup.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+void op::EmbeddingLookup::validate_and_infer_types()
+{
+    element::Type result_et = get_input_element_type(1);
+
+    const PartialShape& arg0_shape = get_input_partial_shape(0);
+    const PartialShape& arg1_shape = get_input_partial_shape(1);
+
+    PartialShape result_shape;
+
+    if (arg0_shape.rank().is_static())
+    {
+        NODE_VALIDATION_ASSERT(
```

Comment:
Still need to move this up, outside the `if`. We should fail in this case even if arg0's rank is dynamic.